### ISSUE TITLE
Changing resolution in tutorial outpost causes errors

### ISSUE THREAD

**Issue by Rokvach** [Regalis11/Barotrauma#10542]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I was testing the reactor infographic branch and noticed that whenever I changed resolution the console was spammed with errors, switched to dev and was able to reproduce the issue there as well

### Reproduction steps

1. Start singleplayer campaign
2. At the outpost change resolution to the lowest
3. Notice that there's errors on the console

### Bug prevalence

Happens every time I play

### Version

Other

### -

dev branch

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
System.ArgumentOutOfRangeException: Index must be within the bounds of the List. (Parameter 'index')
   at System.Collections.Generic.List`1.Insert(Int32 index, T item)
   at Barotrauma.RectTransform.RepositionChildInHierarchy(Int32 index) in C:\Users\RoniK\Documents\Barotrauma-development\Barotrauma\BarotraumaClient\ClientSource\GUI\RectTransform.cs:line 714
```

**Comment by ekqvist** [comment]
Seems to be related to the child objectives, they throw errors and aren't re-added after changing resolution

**Comment by ekqvist** [comment]
Fixed in Regalis11/Barotrauma-development@1d25304

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
